projects/step-frontend/src/lib/os-plugins/modules/script-editor/injectables/authguard-service.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanDeactivate, RouterStateSnapshot } from '@angular/router';
+import { ScriptEditorComponent } from '@exense/step-frontend';
+import { DeactivateComponentDataInterface } from '../types/deactivate-component-data.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuardService implements CanDeactivate<ScriptEditorComponent> {
+  canDeactivate(
+    component: DeactivateComponentDataInterface,
+    currentRoute: ActivatedRouteSnapshot,
+    currentState: RouterStateSnapshot,
+    nextState: RouterStateSnapshot
+  ): boolean {
+    return component.canExit();
+  }
+}
```

Comment:
@lukasz-lepa CanDeactivate guard id deprecated
https://angular.io/api/router/CanDeactivate
Is it possible to replace it with CanDeactivateFn for this case?

Reply:
Changed guard to a CAnDeactivateFn function